### PR TITLE
SED 2915-be-support-new-custom-columns

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

**Review by jeromecomte — CHANGES_REQUESTED** [review]
The PR looks good. Haven't tried to understand the exact logic of SettingScopeRegistry. I would check the test coverage of this class to ensure that everything is properly covered by the unit tests

File: step-controller/step-controller-server/src/main/java/step/migration/tasks/ScreenTemplateMigrationTask25.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	public ScreenTemplateMigrationTask25(CollectionFactory collectionFactory, MigrationContext migrationContext) {
+		super(new Version(3, 25, 0), collectionFactory, migrationContext);
+		screenInputs = collectionFactory.getCollection("screenInputs", Document.class);
+	}
+
+	@Override
+	public void runUpgradeScript() {
+		logger.info("Cleaning up screen inputs which are replaced by the new table and columns configurations mechanism.");
+		screenInputs.remove(Filters.equals("screenId", "executionTable"));
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-controller/step-controller-server/src/main/java/step/migration/tasks/ScreenTemplateMigrationTask25.java:43]: Custom columns settings are currently not supported (i.e. either they are predefined by the FE or they are defined in the screen templates of Execution parameters, plans or functions). 
We could try to modify this migration tasks as part of the related ticket ([SED-3013]). Note that implementing such a migration will require to create the default settings (predefined columns) in the BE

[the remark this answers is not in the document]
Comment (jeromecomte) [reply step-controller/step-controller-server/src/main/java/step/migration/tasks/ScreenTemplateMigrationTask25.java:43]: I thought we could do the following as part of this migration task: For all standard colums (columns that are now predefined in the FE) that are not displayed per default, could we simply display them if a screenInputs corresponding to the column?

For non standard columns I wouldn't do anything but just log a warning in the migration logs

File: step-controller/step-controller-server/src/main/java/step/core/plans/PlanPlugin.java [hunk 2]
```diff
@@ -92,23 +92,6 @@ public void initializeData(GlobalContext context) throws Exception {
 	protected void createScreenInputDefinitionsIfNecessary(GlobalContext context) {
 		// Plan table
 		ScreenInputAccessor screenInputAccessor = context.get(ScreenInputAccessor.class);
```
Comment (jeromecomte) [inline step-controller/step-controller-server/src/main/java/step/core/plans/PlanPlugin.java:94]: @david-stephan doesn't seem to be used anymore

**Review by jeromecomte — DISMISSED** [review]

**Pull request merged**